### PR TITLE
Fixed a bug where not much stories data was downloaded

### PR DESCRIPTION
**Pull request by mplorentz**
When I added [pagination to the home feed](https://github.com/planetary-social/nos/pull/793) (with no PR review lol) I didn't think about the fact that the home feed was also downloading all the events for the stories interface. I did some testing but because I was opening the app frequently I was getting enough recent events to not notice [this bug](https://github.com/planetary-social/nos/issues/767#issuecomment-1893969798).

I was worried I would have to build a whole custom pagination scheme for the home feed, but it turns out just requesting all notes from all our followers from the past couple days isn't that many notes, even if you follow 700 people, so I think the performance on this is good enough, but give it a try and let me know.

### CODE REVIEW

**Comment by mplorentz** [comment]
No CHANGELOG entry because this bug was never released to production.

File: Nos/Views/Home/HomeFeedView.swift [hunk 1]
```diff
@@ -48,6 +48,23 @@ struct HomeFeedView: View {
         )
     }
     
+    /// Downloads the data we need to show stories. 
+    func downloadStories() async {
+        relaySubscriptions.removeAll()
+        
+        let followedKeys = await Array(currentUser.socialGraph.followedKeys)
+        
+        if !followedKeys.isEmpty {
```
Comment (joshuatbrown) [inline Nos/Views/Home/HomeFeedView.swift:57]: I might guard here but I think that's mostly personal preference so the main code isn't nested at all. 🤷

**Review by joshuatbrown — APPROVED** [review]

File: Nos/Views/Home/HomeFeedView.swift (hunk 1 above)
Comment (mplorentz) [reply Nos/Views/Home/HomeFeedView.swift:57]: that might be nice. I actually was just restoring this code from an older commit so I'm going to leave it for now.

**Comment by mplorentz** [comment]
Only the EventObservationTests failed which have been failing on main too, so I'm going to go ahead and merge this.

**Pull request merged**